moshi/src/main/java/com/squareup/moshi/StandardJsonAdapters.java

```diff
@@ -215,6 +223,31 @@ static int rangeCheckNextInt(JsonReader reader, String typeMessage, int min, int
     }
   };
 
+  /**
+   * Loads the generated JsonAdapter for classes annotated {@link JsonClass}. This works because it
+   * uses the same naming conventions as {@code JsonClassCodeGenProcessor}.
+   */
+  static JsonAdapter<?> generatedAdapter(Moshi moshi, Type type, Class<?> rawType) {
+    String adapterClassName = rawType.getName().replace("$", "_") + "JsonAdapter";
+    try {
+      @SuppressWarnings("unchecked") // We generate types to match.
+      Class<? extends JsonAdapter<?>> adapterClass = (Class<? extends JsonAdapter<?>>)
+          Class.forName(adapterClassName, true, rawType.getClassLoader());
+      if (type instanceof ParameterizedType) {
+        return adapterClass.getDeclaredConstructor(Moshi.class, Type[].class)
```

Comment:
Want to fold in the defensive accessibility check from #465 while you're at it?

Reply:
Can do